components/src/core/components/Input/Time/TimePicker.vue

```diff
@@ -170,23 +169,14 @@ export default defineComponent({
     };
 
     const onHourInputBlur = () => {
-      const prevValue = prevHour;
-      const newValue = isValid(enteredValue, 'hour') ? enteredValue : prevValue;
-      prevMinute = newValue;
-
```

Comment:
Calling setValue() is only needed if the new value is valid. 
Calling setValue() with the previous value is not needed, since it should already be set